lib/web/desktop_playback.go

```diff
@@ -170,10 +170,16 @@ func (pp *playbackPlayer) streamSessionEvents(ctx context.Context, cancel contex
 			if !errors.Is(err, context.Canceled) {
 				pp.log.WithError(err).Errorf("streaming session %v", pp.sID)
 			}
+
 			return
 		case evt := <-eventsC:
 			if evt == nil {
 				pp.log.Debug("reached end of playback")
+
+				if _, err := pp.ws.Write([]byte("end")); err != nil {
```

Comment:
Not a fan of breaking protocol here. Everything we're sending now is JSON-encoded protobufs, right?

Can we either:
- send some formatted JSON to signal the end of playback,
- just close the websocket to indicate the end of playback, or
- use the timing data to determine which message is the last one

Reply:
Closing the websocket would be consistent with the behavior of the other `StreamSessionEvents` that we have around the codebase.

Reply:
> just close the websocket to indicate the end of playback, or

@espadolini fyi the websocket does get closed (because the goroutine returns below, which triggers the deferred `defer pp.close(cancel)`.

The reason I don't want to use the websocket closing as the indicator for end of playback on the frontend is that this is not necessarily behavior we'll keep -- eventually we'll want to be able to restart without refreshing the page, in which case we probably want to just keep the websocket open.

> use the timing data to determine which message is the last one

I don't think this is possible, which is the impetus for this change -- the problem is that the last message we play back won't necessarily have a timestamp that's congruent with the total time of the recording according to the `WindowsDesktopSessionEnd` event. For example, the last message might be a PNG, and then the user might sit there staring at the screen for 30 seconds before closing the tab. Because we're using the `WindowsDesktopSessionEnd` event's timing data to determine the total time of the recording on the frontend, our progress bar will just stop with 30 seconds to go, and we need some way to know that it should be carried through to the end.

> send some formatted JSON to signal the end of playback

Barring being told I'm missing something above, I'll go with this.

Reply:
Ahh, thank you - that makes a lot of sense. I hadn't considered idle time at the end.

The approach seems fine then, I would just make sure everything we send on the playback is JSON so the frontend doesn't have to handle different formats.